### PR TITLE
[fix](vault) Fix bugs about altering storage vault name

### PR DESCRIPTION
**Pull request by swjtu-zhanglei**
* fix altering storage name but not writing disk in meta-service
* check vault if existed when altering stoarge vault name

### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [ ] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by hello-stephen** [comment]
Thank you for your contribution to Apache Doris.
Don't know what should be done next? See [How to process your PR](https://cwiki.apache.org/confluence/display/DORIS/How+to+process+your+PR).

Please clearly describe your PR:
1. What problem was fixed (it's best to include specific error reporting information). How it was fixed.
2. Which behaviors were modified. What was the previous behavior, what is it now, why was it modified, and what possible impacts might there be.
3. What features were added. Why was this function added?
4. Which code was refactored and why was this part of the code refactored?
5. Which functions were optimized and what is the difference before and after the optimization?

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Comment by SWJTU-ZhangLei** [comment]
run buildall

File: gensrc/proto/cloud.proto [hunk 1]
```diff
@@ -887,6 +887,9 @@ message AlterObjStoreInfoResponse {
     optional MetaServiceResponseStatus status = 1;
     optional string storage_vault_id = 2;
     optional bool default_storage_vault_replaced = 3;
+
+    // storage_vault_name maybe changed, so return new storage_vault_name
+    optional string storage_vault_name = 4;
```
Comment (gavinchou) [inline gensrc/proto/cloud.proto:892]: redundant vault name, using the name in request

File: cloud/src/meta-service/meta_service_resource.cpp [hunk 2]
```diff
@@ -523,9 +523,18 @@ static void set_default_vault_log_helper(const InstanceInfoPB& instance,
...
+static bool vault_exist(const InstanceInfoPB& instance, const std::string& new_vault_name) {
+    for (auto& name : instance.storage_vault_names()) {
+        if (new_vault_name == name) {
+            return true;
+        }
+    }
+    return false;
+}
+
+static int alter_hdfs_storage_vault(InstanceInfoPB& instance, std::unique_ptr<Transaction>& txn,
```
Comment (gavinchou) [inline cloud/src/meta-service/meta_service_resource.cpp:535]: add UT

**Comment by SWJTU-ZhangLei** [comment]
run buildall

File: cloud/src/meta-service/meta_service_resource.cpp (hunk 2 above)
Comment (SWJTU-ZhangLei) [reply cloud/src/meta-service/meta_service_resource.cpp:535]: > add UT

done

File: gensrc/proto/cloud.proto (hunk 1 above)
Comment (SWJTU-ZhangLei) [reply gensrc/proto/cloud.proto]: > redundant vault name, using the name in request

done

**Review by gavinchou — APPROVED** [review]

File: regression-test/suites/vault_p0/alter/test_alter_vault_name.groovy [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+    // case1
+    expectExceptionLike({
+        sql """
+            ALTER STORAGE VAULT ${hdfsVaultName}
+            PROPERTIES (
+                "type" = "hdfs",
+                "VAULT_NAME" = "${hdfsVaultName}"
+            );
+        """
+    }, "vault name no change")
```
Comment (MoanasDaddyXu) [inline regression-test/suites/vault_p0/alter/test_alter_vault_name.groovy:68]: vault name no change -> unable to use the original vault name?

**Review by gavinchou — APPROVED** [review]

**Review by MoanasDaddyXu — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**